server/api/utils/singletons/k8s.py

```diff
@@ -125,6 +125,115 @@ def list_pods(self, namespace=None, selector="", states=None):
                 items.append(i)
         return items
 
+    @raise_for_status_code
+    def list_pods_paginated(self, namespace=None, selector="", states=None):
+        _continue = ""
+        logger.debug(
+            "Listing namespaced pods with pagination",
+            selector=selector,
+            namespace=namespace,
+            states=states,
+        )
+        while True:
+            if not _continue:
+                pods_list = self.v1api.list_namespaced_pod(
+                    self.resolve_namespace(namespace),
+                    label_selector=selector,
+                    watch=False,
+                    limit=200,
+                )
+            else:
+                pods_list = self.v1api.list_namespaced_pod(
+                    self.resolve_namespace(namespace),
+                    label_selector=selector,
+                    watch=False,
+                    limit=200,
+                    _continue=_continue,
+                )
+
+            for item in pods_list.items:
+                if not states or item.status.phase in states:
+                    yield item
+
+            _continue = pods_list._metadata._continue
```

Comment:
No better way to resolve the continue?

Reply:
metadata has a public getter but continue doesn't
also still checking how this works for crds